patches/components-permissions-android-BUILD.gn.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/components/permissions/android/BUILD.gn b/components/permissions/android/BUILD.gn
+index 3cfa933430a16ea55bfae73aa02e195a0aedd5a9..79d9bfcd28fa8b1c490213cb34abd31943a64e88 100644
+--- a/components/permissions/android/BUILD.gn
++++ b/components/permissions/android/BUILD.gn
+@@ -14,6 +14,7 @@ generate_jni("jni_headers") {
+     "java/src/org/chromium/components/permissions/PermissionUtil.java",
+     "java/src/org/chromium/components/permissions/nfc/NfcSystemLevelSetting.java",
+   ]
++  sources += brave_components_permissions_java_jni_sources
```

Comment:
do we need these individual source patches for java? It seems like they are coming up a lot recently and it's not clear to me if this is strictly necessary like it is for cpp or if they can all be added in the same place somewhere else

Reply:
@AlexeyBarabash said he would look at this tomorrow

Reply:
@goodov , @bridiver,

Yes, it is possible to eliminate `brave_components_permissions_java_jni_sources` and one line of patch.
`generate_jni("jni_headers")` target in fact makes one textfile from another and does depend on java-files only, no external dependancies or other requirements. But on `jni_headers` depends the corresponding .cc file which includes the generated header.

So, here is what could be done:
1. Introduce `src/brave/components/permissions/android/BUILD.gn` with `generate_jni("jni_headers")` target
2. Remove `brave_components_permissions_java_jni_sources`  from `patches/components-permissions-android-BUILD.gn.patch` and `components/permissions/android/java_sources.gni`
3. re-use patch line at `patches/components-permissions-BUILD.gn.patch`
```
-+  import("//brave/components/permissions/sources.gni") sources += brave_components_permissions_sources deps += brave_components_permissions_deps
++  import("//brave/components/permissions/sources.gni") sources += brave_components_permissions_sources deps += brave_components_permissions_deps if (is_android) { deps += [ "//brave/components/permissions/android:jni_headers" ] }
```
4. Modify header path in `chromium_src/components/permissions/android/permission_dialog_delegate.cc`
```
-#include "components/permissions/android/jni_headers/BravePermissionDialogDelegate_jni.h"
+#include "brave/components/permissions/android/jni_headers/BravePermissionDialogDelegate_jni.h"
```

Reply:
@AlexeyBarabash thank you!